profiler/README.md

```diff
@@ -1,9 +1,46 @@
 
-The profiler is in its infancy and is disabled by default.
+🚨 The profiler is in beta and its use is currently restricted.
```

Comment:
We usually add
```markdown
> :construction: &nbsp;Status: Experimental
```
to beta/experimental features' docs; see [metrics.md](https://github.com/signalfx/splunk-otel-java/blob/main/docs/metrics.md) for an example.

Reply:
Sounds good. 👍🏻